### PR TITLE
Inline `shallow_resolve_ty` into `ShallowResolver`

### PR DESCRIPTION
**Pull request by compiler-errors**
addresses fixme I found in infcx

### CODE REVIEW

File: compiler/rustc_infer/src/infer/mod.rs [hunk 1]
```diff
@@ -1831,8 +1788,48 @@ impl<'a, 'tcx> TypeFolder<'tcx> for ShallowResolver<'a, 'tcx> {
...
+            }
+
+            ty::Infer(ty::IntVar(v)) => self
+                .infcx
+                .inner
+                .borrow_mut()
+                .int_unification_table()
+                .probe_value(v)
+                .map(|v| v.to_type(self.infcx.tcx))
+                .unwrap_or(ty),
```
Comment (klensy) [inline compiler/rustc_infer/src/infer/mod.rs:1820]: Maybe `map_or`?

File: compiler/rustc_infer/src/infer/mod.rs [hunk 2]
```diff
@@ -1831,8 +1788,48 @@ impl<'a, 'tcx> TypeFolder<'tcx> for ShallowResolver<'a, 'tcx> {
...
+
+            ty::Infer(ty::FloatVar(v)) => self
+                .infcx
+                .inner
+                .borrow_mut()
+                .float_unification_table()
+                .probe_value(v)
+                .map(|v| v.to_type(self.infcx.tcx))
+                .unwrap_or(ty),
```
Comment (klensy) [inline compiler/rustc_infer/src/infer/mod.rs:1829]: The same.

Comment (compiler-errors) [reply compiler/rustc_infer/src/infer/mod.rs:1829]: sure, literally just copied code around, but it doesn't hurt to also fix it up a bit, haha

**Pull request merged**